### PR TITLE
Remove javassist from agent dependency

### PR DESCRIPTION
**Pull request by wu-sheng**
Through @raphw helps in these days, I got the ideas of how to use ByteBuddy only to do bootstrap instrumentation. So, I remove the javassist from the agent side dependency, and from my tests, all features are still there.

New in this PR, add several byte buddy annotation classes(used by us) into the bootstrap class loader.

@candyleer As you are doing the instrumentation of jdk HTTP Connection, please verify.

FYI All @apache/skywalking-committers, javassist will be gone from the agent. Only depend on ByteBuddy again, at least we wouldn't need to have concerns about that, javassist was no longer supported in JDK6 for at least a year ago.

### CODE REVIEW

**Comment by candyleer** [comment]
wow， i will verify today

**Comment by wu-sheng** [comment]
Any update? @candyleer

**Comment by candyleer** [comment]
wait a moment ,I am verifying now

**Comment by candyleer** [comment]
verified .works well

**Comment by wu-sheng** [comment]
Thanks. Please approve and merge.

File: apm-sniffer/apm-agent-core/src/main/java/org/apache/skywalking/apm/agent/core/plugin/bootstrap/BootstrapInstrumentBoost.java [hunk 1]
```diff
@@ -160,29 +173,49 @@ private static boolean prepareJREInstrumentation(PluginFinder pluginFinder,
...
+            DynamicType.Unloaded interceptorType = new ByteBuddy()
+                .redefine(templateTypeDescription, ClassFileLocator.ForClassLoader.of(BootstrapInstrumentBoost.class.getClassLoader()))
+                .name(internalInterceptorName)
+                .field(named("TARGET_INTERCEPTOR")).value(methodsInterceptor)
+                .make();
+
+            classesTypeMap.put(internalInterceptorName, interceptorType.getBytes());
+
+            try {
+                File debuggingClassesRootPath = null;
```
Comment (candyleer) [inline apm-sniffer/apm-agent-core/src/main/java/org/apache/skywalking/apm/agent/core/plugin/bootstrap/BootstrapInstrumentBoost.java]: wirte debug class when Config.Agent.IS_OPEN_DEBUGGING_CLASS is true?

Comment (wu-sheng) [reply apm-sniffer/apm-agent-core/src/main/java/org/apache/skywalking/apm/agent/core/plugin/bootstrap/BootstrapInstrumentBoost.java]: Yes. It exists before, I just move it to support this new part.

Comment (candyleer) [reply apm-sniffer/apm-agent-core/src/main/java/org/apache/skywalking/apm/agent/core/plugin/bootstrap/BootstrapInstrumentBoost.java]: Yeah,I mean I think it is not necessary to write the class to disk every time and It seems duplicated in this place.
image

Comment (wu-sheng) [reply apm-sniffer/apm-agent-core/src/main/java/org/apache/skywalking/apm/agent/core/plugin/bootstrap/BootstrapInstrumentBoost.java]: I will remove the duplicated ones.

Comment (wu-sheng) [reply apm-sniffer/apm-agent-core/src/main/java/org/apache/skywalking/apm/agent/core/plugin/bootstrap/BootstrapInstrumentBoost.java]: I removed the duplicated file write and map#put. Please recheck, if it is OK now, please approve again.

**Pull request merged**